### PR TITLE
Gstreamer1.0

### PR DESCRIPTION
**Pull request by alarcher**
Introspection is broken for multilib builds... needs work.

### CODE REVIEW

File: components/desktop/gst-plugins-base1/gst-plugins-base1.p5m [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+<transform file path=usr/share/locale/([^/]+)(\..+){0,1}(/.+){0,1} -> default facet.locale.%<\1> true>
```
Comment (xen0l) [inline components/desktop/gst-plugins-base1/gst-plugins-base1.p5m]: This transform is not needed as we have it here: https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/transforms/locale#L32

File: components/desktop/gst-plugins-base1/gst-plugins-base1.p5m [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+file path=usr/include/gstreamer-1.0/gst/video/video-info.h
+file path=usr/include/gstreamer-1.0/gst/video/video-multiview.h
+file path=usr/include/gstreamer-1.0/gst/video/video-overlay-composition.h
+file path=usr/include/gstreamer-1.0/gst/video/video-resampler.h
+file path=usr/include/gstreamer-1.0/gst/video/video-scaler.h
+file path=usr/include/gstreamer-1.0/gst/video/video-tile.h
+file path=usr/include/gstreamer-1.0/gst/video/video.h
+file path=usr/include/gstreamer-1.0/gst/video/videoorientation.h
+file path=usr/include/gstreamer-1.0/gst/video/videooverlay.h
+#file path=usr/lib/$(MACH64)/girepository-1.0/GstAllocators-1.0.typelib
```
Comment (xen0l) [inline components/desktop/gst-plugins-base1/gst-plugins-base1.p5m]: Is this related to broken multilib?

File: components/desktop/gst-plugins-good1/gst-plugins-good1.p5m [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+<transform file path=usr/share/locale/([^/]+)(\..+){0,1}(/.+){0,1} -> default facet.locale.%<\1> true>
```
Comment (xen0l) [inline components/desktop/gst-plugins-good1/gst-plugins-good1.p5m]: You can omit this line. We have it here: https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/transforms/locale#L32

File: components/desktop/gst-plugins-base1/gst-plugins-base1.p5m (hunk 2 above)
Comment (alarcher) [reply components/desktop/gst-plugins-base1/gst-plugins-base1.p5m]: Gobject introspection is broken for 32bit multilib, some types seem to be defined as 64bit which causes overflows: I could not find the reason.
Several upstream bug reports by Arch Linux were closed without any proper conclusion.

File: components/desktop/gst-plugins-base1/patches/01.filio.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+--- gst-plugins-base-1.8.1/gst/tcp/gstmultifdsink.c.orig	2016-09-05 18:52:34.896956799 +0200
++++ gst-plugins-base-1.8.1/gst/tcp/gstmultifdsink.c	2016-09-05 18:54:45.854211639 +0200
+@@ -118,6 +118,9 @@
+ #include <sys/stat.h>
+ #include <netinet/in.h>
+ 
++#if defined (__sun) && defined(__SVR4)
++#define HAVE_FIONREAD_IN_SYS_FILIO
```
Comment (pyhalov) [inline components/desktop/gst-plugins-base1/patches/01.filio.patch]: Shouldn't this be detected by configure? If so, why wasn't it detected?

Comment (alarcher) [reply components/desktop/gst-plugins-base1/patches/01.filio.patch]: I do not know...
Considering that our Gobject introspection is broken with multilib would you mind merging with either disabling introspection for 32bit of both?

File: components/multimedia/gstreamer1/Makefile [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+COMPONENT_ARCHIVE_URL= \
+  http://gstreamer.freedesktop.org/src/$(COMPONENT_SRC_NAME)/$(COMPONENT_ARCHIVE)
+COMPONENT_LICENSE= LGPLv2
+COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+gcc_OPT = -O2
```
Comment (pyhalov) [inline components/multimedia/gstreamer1/Makefile:38]: Interesting, if it can be removed

File: components/desktop/gst-plugins-base1/gst-plugins-base1.p5m (hunk 2 above)
Comment (pyhalov) [reply components/desktop/gst-plugins-base1/gst-plugins-base1.p5m]: Worked in my limited testing (built gstreamer 1, launched some gst-launch-1.0, gst-typefind-1.0 and so on. Would be interesting to find out how to reproduce the issue with introspection.

**Comment by alarcher** [comment]
@pyhalov Ready for merge. Plugins to follow.
